Reachable unwrap panic in `VsockStreamSocket::drop`

### Describe the bug
There is a reachable unwrap panic in `VsockStreamSocket::drop()` at kernel/src/net/socket/vsock/stream/socket.rs:306 when closing socket.

https://github.com/asterinas/asterinas/blob/525085ab865b3850c0de2d704bab1bba6976a980/kernel/src/net/socket/vsock/stream/socket.rs#L306

### To Reproduce
1. Compile the program to call `socket` and close it.
```C
#include <stdio.h>
#include <stdlib.h>
#include <sys/socket.h>
#include <linux/vm_sockets.h>
#include <unistd.h>
#include <errno.h>

int main() {
    int fd = socket(AF_VSOCK, SOCK_STREAM, 0);
    if (fd < 0) {
        perror("socket(AF_VSOCK) failed");
        return 1;
    }
    close(fd);
    printf("Closed vsock socket\n");
    return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior

Asterinas shall not panic.

### Environment
- Official docker asterinas/asterinas:0.15.0-20250516
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 525085a

### Logs
```
~ # /root/socket.c 
[  5014.010] ERROR: Uncaught panic:
        called `Option::unwrap()` on a `None` value
        at /root/asterinas/kernel/src/net/socket/vsock/stream/socket.rs:306
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, status: AtomicThreadStatus(1), cpu_affinity: AtomicCpuSet { bits: [1] }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31490640 }, fair: FairAttr { weight: 1024, vruntime: 6286077796 } } })
Printing stack trace:
   1: fn 0xffffffff88828c30 - pc 0xffffffff88828c7a / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0020;

   2: fn 0xffffffff8804cbc0 - pc 0xffffffff8804d369 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0070;

   3: fn 0xffffffff8804cbb0 - pc 0xffffffff8804cbba / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0480;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0490;

   5: fn 0xffffffff88942720 - pc 0xffffffff88942764 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb04a0;

   6: fn 0xffffffff889427d0 - pc 0xffffffff88942810 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb04e0;

   7: fn 0xffffffff8894d240 - pc 0xffffffff8894d255 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0550;

   8: fn 0xffffffff8854b5e0 - pc 0xffffffff8854b629 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0560;

   9: fn 0xffffffff88532100 - pc 0xffffffff88532113 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb06b0;

  10: fn 0xffffffff88530150 - pc 0xffffffff8853017d / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb06d0;

  11: fn 0xffffffff884fafc0 - pc 0xffffffff884fb012 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0700;

  12: fn 0xffffffff88534ab0 - pc 0xffffffff88534b33 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0740;

  13: fn 0xffffffff885325f0 - pc 0xffffffff885325fa / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb07c0;

  14: fn 0xffffffff880c5480 - pc 0xffffffff880c5498 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb07d0;

  15: fn 0xffffffff881a22a0 - pc 0xffffffff881a250f / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb07f0;

  16: fn 0xffffffff88361840 - pc 0xffffffff88364943 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedb0980;

  17: fn 0xffffffff8804c820 - pc 0xffffffff8804c8b1 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2510;

  18: fn 0xffffffff8816a720 - pc 0xffffffff8816aefa / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2690;

  19: fn 0xffffffff885237b0 - pc 0xffffffff885237b6 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2d00;

  20: fn 0xffffffff883d4730 - pc 0xffffffff883d4747 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2d10;

  21: fn 0xffffffff883d4690 - pc 0xffffffff883d46b1 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2d30;

  22: fn 0xffffffff8804b200 - pc 0xffffffff8804b212 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2d80;

  23: fn 0xffffffff882fda80 - pc 0xffffffff882fda96 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2db0;

  24: fn 0xffffffff883f1f80 - pc 0xffffffff883f1f8c / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2f50;

  25: fn 0xffffffff88522830 - pc 0xffffffff8852283e / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2f80;

  26: fn 0xffffffff887eb270 - pc 0xffffffff887eb295 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2fa0;

  27: fn 0xffffffff887e68d0 - pc 0xffffffff887e6969 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc2fd0;

  28: fn                0x0 - pc 0xffff80027310d620 / registers:

     rax                0x0; rdx 0xffffdffffedaff88; rcx 0xffffdffffedaff10; rbx                0x0;
     rsi 0xffffffff88967a70; rdi 0xffff800272c1d558; rbp                0x0; rsp 0xffffdffffedc3000;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panic.rs:107
(  2) /root/asterinas/kernel/src/thread/oops.rs:129
(  3) ??:?
(  4) bb0e31ooocbssppv6q3sbsuao:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:217
(  7) ??:?
(  8) ??:?
(  9) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:523
( 10) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:523
( 11) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/sync.rs:1910
( 12) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/sync.rs:2597
( 13) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:523
( 14) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/mem/mod.rs:940
( 15) /root/asterinas/kernel/src/syscall/close.rs:34
( 16) /root/asterinas/kernel/src/syscall/mod.rs:180
( 17) /root/asterinas/kernel/src/syscall/mod.rs:351
( 18) /root/asterinas/kernel/src/thread/task.rs:83
( 19) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 20) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:56
( 21) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:42
( 22) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panic.rs:87
( 23) /root/asterinas/kernel/src/thread/oops.rs:54
( 24) /root/asterinas/kernel/src/thread/task.rs:109
( 25) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 26) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2016
( 27) /root/asterinas/ostd/src/task/mod.rs:234
( 28) ??:0
make: *** [Makefile:230: run] Error 1
```

--- a/kernel/src/net/socket/vsock/stream/socket.rs
+++ b/kernel/src/net/socket/vsock/stream/socket.rs
@@ -29,12 +29,19 @@ pub enum Status {
 }
 
 impl VsockStreamSocket {
-    pub fn new(nonblocking: bool) -> Self {
+    pub fn new(nonblocking: bool) -> Result<Self> {
+        if VSOCK_GLOBAL.get().is_none() {
+            return_errno_with_message!(
+                Errno::EINVAL,
+                "cannot create vsock socket since no vsock device is found"
+            );
+        }
+
         let init = Arc::new(Init::new());
-        Self {
+        Ok(Self {
             status: RwLock::new(Status::Init(init)),
             is_nonblocking: AtomicBool::new(nonblocking),
-        }
+        })
     }
 
     pub(super) fn new_from_connected(connected: Arc<Connected>) -> Self {

--- a/kernel/src/syscall/socket.rs
+++ b/kernel/src/syscall/socket.rs
@@ -77,7 +77,7 @@ pub fn sys_socket(domain: i32, type_: i32, protocol: i32, ctx: &Context) -> Resu
             }
         }
         (CSocketAddrFamily::AF_VSOCK, SockType::SOCK_STREAM) => {
-            Arc::new(VsockStreamSocket::new(is_nonblocking)) as Arc<dyn FileLike>
+            Arc::new(VsockStreamSocket::new(is_nonblocking)?) as Arc<dyn FileLike>
         }
         _ => return_errno_with_message!(Errno::EAFNOSUPPORT, "unsupported domain"),
     };

